handlers/handler-publickey-auth/src/main/java/io/servicecomb/authentication/consumer/ConsumerAuthHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 Huawei Technologies Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicecomb.authentication.consumer;
+
+import java.util.Optional;
+
+import io.servicecomb.core.Const;
+import io.servicecomb.core.Handler;
+import io.servicecomb.core.Invocation;
+import io.servicecomb.swagger.invocation.AsyncResponse;
+
+/**
+ * 
+ * add token to context
+ * Provider will get token for authentication
+ *
+ */
+public class ConsumerAuthHandler implements Handler {
+
+  private RSAConsumerTokenManager athenticationTokenManager = new RSAConsumerTokenManager();
+
+  @Override
+  public void handle(Invocation invocation, AsyncResponse asyncResp) throws Exception {
+
+    Optional<String> token = Optional.ofNullable(athenticationTokenManager.getToken());
+    if(!token.isPresent())
```

Comment:
I think, we can delete all these instructions.

Reply:
maybe,but i think can reduce send request to server if do it